Libraries/LinkingIOS/RCTLinkingManager.m

```diff
@@ -42,18 +42,31 @@ - (void)stopObserving
   return @[@"url"];
 }
 
++ (BOOL)application:(UIApplication *)app
+            openURL:(NSURL *)URL
+            options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options
+{
+    [self postNotificationWithURL:URL];
+    return YES;
+}
+
 + (BOOL)application:(UIApplication *)application
             openURL:(NSURL *)URL
   sourceApplication:(NSString *)sourceApplication
          annotation:(id)annotation
 {
-  NSDictionary<NSString *, id> *payload = @{@"url": URL.absoluteString};
-  [[NSNotificationCenter defaultCenter] postNotificationName:RCTOpenURLNotification
-                                                      object:self
-                                                    userInfo:payload];
+  [self postNotificationWithURL:URL];
   return YES;
 }
 
++ (void)postNotificationWithURL:(NSURL *)URL
```

Comment:
@javache k updated, had to add `id class;` at the top because unable to call `self` in c function